modules/extension/wmts/src/main/java/org/geotools/ows/wmts/WebMapTileServer.java

```diff
@@ -185,6 +195,12 @@ protected void setupSpecifications() {
         return tileRequest.getTiles();
     }
 
+    /** The new issueRequest, which gives a GetTileResponse in response for a GetTileRequest. */
+    public GetTileResponse issueRequestForTileResponse(GetTileRequest tileRequest)
```

Comment:
I believe this new API is confusing, in the context of what GetTileRequest can do... the object has a generic bounding box and will fetch al the tiles in it. It also allows generic width/height. The request must be carefully built to just get one tile. It seems to me, if you wanted to fetch a single tile, then a new request/response pairs should be used instead, and make sure they can only do that (e.g., identify the tile by xyz rather than bbox/with/height).

Reply:
For me the old API have been confusing since I started to look into this. The GetTileResponse wasn't actually used and GetTileRequest was actually many http-calls without using the ordinary OWS-mechanism. So I doubt it would handle an error from the server, and surely not vector tiles.

I was thinking about adding the properties for tileset, row and col. Was just wondering how to do it. I could perhaps do it with a parallell interface, but that seems like a lot of work.

Reply:
I agree the API is confusing (an operation that returns multiple tiles should be called GetTiles), but the fix does not make it better, since the parameters of GetTileRequest are still allowing for multiple tiles.

A potential way out of this without a solid API break can be:
- Rename GetTile to getMultiTile
- Create a deprecated subclass of it called GetTile
- Create a getSingleTile request whose parameters only allow to ask for a single tile (e.g. x,y,z)

Also, a note: asking for multiple tiles is not a weird operation, while it's not in the WMTS standard, it has been considered for addition for a long time, just did not happen due to lack of resources in OGC.
OGC API Tiles instead allows to request single tiles in core, while allowing for multiple tiles in [an optional extension](https://github.com/opengeospatial/ogcapi-tiles/blob/master/extensions/multitile/standard/clause_7_tile_multitiles.adoc).

The reason why multiple tiles request should be allowed is evident, it's more efficient, instead of paying latency N times, it will be paid only once.

Reply:
I've introduced GetSingleTileRequest, GetMultiTileRequest and also GetKVPTileRequest and GetRestTileRequest, and deprecated the old GetTileRequest.

I see the point of getting many tiles by one http call, but anyway there would have been a Response that wrapped the http response to extract those tiles. Not like today where it comes from a call to WMTSTileService.